drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -124,6 +126,14 @@ static int i2c_nrfx_twim_transfer(const struct device *dev,
 		cur_xfer.type = (msgs[i].flags & I2C_MSG_READ) ?
 			NRFX_TWIM_XFER_RX : NRFX_TWIM_XFER_TX;
 
+		if (cur_xfer.primary_length > I2C_NRFX_MAX_TRANSFER_SIZE) {
+			LOG_ERR("Trying to transfer more than the maximum size "
+				"for this device: %d > %d",
+				cur_xfer.primary_length,
+				I2C_NRFX_MAX_TRANSFER_SIZE);
+			return -ENOSPC;
+		}
```

Comment:
I think you should introduce `easydma-maxcnt-bits` devicetree property, as it's done for SPI. Even if current SoCs do share the same value for all instances, it should not be assumed this will always be the case.

Reply:
oh that makes sense, did not realize it was a thing already on SPI, yeah I'll do that

Reply:
Done